### PR TITLE
Informative not found error

### PR DESCRIPTION
**Pull request by pyakushin**
Contains changes from #46 PR

### CODE REVIEW

File: src/core_inner/backend/pearl/group.rs [hunk 1]
```diff
@@ -242,22 +242,18 @@ impl PearlGroup {
...
             }
         }
-        if results.is_empty() {
-            if has_error {
-                debug!("cannot read from some pearls");
-                Err(Error::Failed("cannot read from some pearls".to_string()))
-            } else {
-                Err(Error::KeyNotFound)
-            }
+        if results.is_empty() && has_error {
```
Comment (agend) [inline src/core_inner/backend/pearl/group.rs]: Why error and empty results branches put together?

File: src/core_inner/cluster/simple.rs [hunk 2]
```diff
@@ -97,26 +97,29 @@ impl Cluster for Quorum {
...
-                }
-            })
-            .boxed();
+        let task = async move {
+            let results = reqs.collect::<Vec<_>>().await;
+            let err_results = results
+                .iter()
+                .filter_map(|r| r.as_ref().err())
+                .collect::<Vec<_>>();
+            let errors = err_results
```
Comment (agend) [inline src/core_inner/cluster/simple.rs]: errors only logs, should we do something useful with them?

File: src/core_inner/backend/pearl/group.rs (hunk 1 above)
Comment (pyakushin) [reply src/core_inner/backend/pearl/group.rs]: I preserve the same logic. Settings::choose data returns result value with only KeyNotFound error. I changed return type to option and did equal logic transormation

File: src/core_inner/cluster/simple.rs (hunk 2 above)
Comment (pyakushin) [reply src/core_inner/cluster/simple.rs]: I've thought, that it is the task for the next refactoring PR. There are a lot of that kind of code

**Pull request merged**